### PR TITLE
Fixes airlocks in maintenance having walls in them

### PR DESCRIPTION
**Pull request by silicons**
*sighs at you*

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -132,10 +132,13 @@
 		switch(outcome)
 			if(1 to 9)
 				var/turf/here = get_turf(src)
+				var/newtype = /turf/closed/wall
 				for(var/turf/closed/T in range(2, src))
-					here.ChangeTurf(T.type)
-					return INITIALIZE_HINT_QDEL
```
Comment (kevinz000) [inline code/game/machinery/doors/airlock.dm]: lateinit doesnt care about your qdel hints.

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -132,10 +132,13 @@
...
-				here.ChangeTurf(/turf/closed/wall)
+					newtype = T.type
+					break
```
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm]: use locate()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Bawhoppen** [comment]
This was an intentional feature. Talk to @Robustin

**Comment by Cyberboss** [comment]
@Bawhoppen the use of the initialize hint says it's not

**Comment by ShizCalev** [comment]
This was intentional #31312

**Comment by ShizCalev** [comment]
image
kek

**Comment by Bawhoppen** [comment]
I swear to god

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (Cyberboss) [reply code/game/machinery/doors/airlock.dm]: and to keep on par with robustin's airlock tweak only qdel it if the locate succeeds

**Comment by ShizCalev** [comment]
conflict.

**Comment by Robustin** [comment]
It was intentional. 

The entire purpose was to have airlocks in maint and other "abandoned" areas obscure whether the door was broken or tampered from actual crew sabotage.

The original behavior included an option that flat out deleted the door and put a wall in its place... except I realized after seeing it live that nobody would actually be convinced because only an engineer would even have the tools to bother with deconstructing a door and replacing it with a wall... and that engineers would never do that when they can just replace the door with one using heightened access requirements. 

What I had seen is people building walls inside an open door to keep prying eyes out. I even paid $500 to commission this state of the art visual aid to help you smoothbrains understand: 

https://thumbs.dreamstime.com/z/brick-wall-behind-open-door-17766141.jpg

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -132,10 +132,12 @@
 		switch(outcome)
 			if(1 to 9)
 				var/turf/here = get_turf(src)
-				for(var/turf/closed/T in range(2, src))
-					here.ChangeTurf(T.type)
-					return INITIALIZE_HINT_QDEL
-				here.PlaceOnTop(/turf/closed/wall)
+				var/turf/closed/T = locate() in range(2, src)
+				var/newtype = T.type
+				if(newtype)
```
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm]: What do you think will happen if T is null @kevinz000 >:[

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ChangelingRain** [comment]
This was literally intended. That's about it.

**Comment by Bawhoppen** [comment]
I was right!

**Comment by Cyberboss** [comment]
The INITIALIZE_HINT_QDEL is still factually wrong

File: code/game/machinery/doors/airlock.dm [hunk 4]
```diff
@@ -132,10 +132,12 @@
...
+					here.ChangeTurf(newtype)
+					qdel(src)
+					return
 			if(9 to 11)
```
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm]: else here.PlaceOnTop(/turf/closed/wall)

File: code/game/machinery/doors/airlock.dm [hunk 5]
```diff
@@ -132,10 +132,12 @@
...
-				for(var/turf/closed/T in range(2, src))
-					here.ChangeTurf(T.type)
-					return INITIALIZE_HINT_QDEL
-				here.PlaceOnTop(/turf/closed/wall)
+				var/turf/closed/T = locate() in range(2, src)
+				var/newtype = T.type
+				if(newtype)
+					here.ChangeTurf(newtype)
+					qdel(src)
+					return
```
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm]: Remove this return

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/airlock.dm [hunk 6]
```diff
@@ -132,10 +132,12 @@
...
 				var/turf/here = get_turf(src)
-				for(var/turf/closed/T in range(2, src))
-					here.ChangeTurf(T.type)
-					return INITIALIZE_HINT_QDEL
-				here.PlaceOnTop(/turf/closed/wall)
+				var/turf/closed/T = locate() in range(2, src)
+				var/newtype = T.type
+				if(newtype)
+					here.ChangeTurf(newtype)
+					qdel(src)
```
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm]: Remove the qdel

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by kevinz000** [comment]
On vacation and my laptop is depowrred.

**Comment by PKPenguin321** [comment]
walls under airlocks is the stupidest intentional feature i've ever seen

**Comment by Bawhoppen** [comment]
>just now people are realising how dumb it is

**Pull request closed without merging**